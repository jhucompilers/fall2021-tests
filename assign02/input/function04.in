// demonstration of scope

var a, b, c;

a = 1;
b = 2;
c = 3;

function add1(a) {
  var b;
  b = 1;
  c = 4;
  a + b;
}

var d;
d = add1(c);

println(a);
println(b);
println(c);

d;
