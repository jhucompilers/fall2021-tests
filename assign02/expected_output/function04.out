1
2
4
Result: 4
